Source/Wit/Public/Wit/Utilities/WitHelperUtilities.cpp

```diff
@@ -321,54 +322,92 @@ FString FWitHelperUtilities::GetVoiceClipId(const FTtsConfiguration& ClipSetting
  * @param RawDataSize [in] the raw data size
  * @return the sound wave created or null if unsuccessful
  */
-USoundWave* FWitHelperUtilities::CreateSoundWaveFromRawData(const uint8* RawData, const int32 RawDataSize)
+USoundWave* FWitHelperUtilities::CreateSoundWaveFromRawData(
+	const uint8* RawData,
+	const int32 RawDataSize,
+	const EWitRequestAudioFormat AudioFormat,
+	const bool bStream)
 {
-	FWaveModInfo WaveInfo;
-	const bool bIsValidWaveInfo = WaveInfo.ReadWaveInfo(RawData, RawDataSize);	
-
-	if (!bIsValidWaveInfo)
+	FSoundWaveParams SoundWaveParams = FSoundWaveParams();
+	USoundWave* SoundWave;
+	switch (AudioFormat)
+	{
+	case EWitRequestAudioFormat::Wav:
 	{
+		FWaveModInfo WaveInfo;
+		const bool bIsValidWaveInfo = WaveInfo.ReadWaveInfo(RawData, RawDataSize);
+
+		if (!bIsValidWaveInfo)
+		{
+			return nullptr;
+		}
+
+		const int32 ChannelCount = *WaveInfo.pChannels;
+		const bool bIsMonoOrStereo = ChannelCount > 0 && ChannelCount <= 2;
+
+		if (!bIsMonoOrStereo)
+		{
+			return nullptr;
+		}
+
+		const int32 SizeOfSample = (*WaveInfo.pBitsPerSample) / 8;
+		const int32 NumSamples = WaveInfo.SampleDataSize / SizeOfSample;
+		const int32 NumFrames = NumSamples / ChannelCount;
+		const bool bIsEmpty = NumFrames <= 0;
+
+		if (bIsEmpty)
+		{
+			return nullptr;
+		}
+
+		const float Duration = static_cast<float>(NumFrames) / *WaveInfo.pSamplesPerSec;
```

Comment:
Can we have the unit suffix to all the duration variable or function, if it's a plain number. For example, DurationSeconds or DurationSec.

Reply:
Sure can add that, was following the same naming convention from `USoundWave`, but no reason our code can't be more specific